main.cc

```diff
@@ -1501,7 +1500,7 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             }
 
             // Delay listening messaging_service until gossip message handlers are registered
-            messaging.start(mscfg, scfg, creds, std::ref(feature_service), std::ref(gossip_address_map)).get();
+            messaging.start(std::ref(sl_controller), mscfg, scfg, creds, std::ref(feature_service), std::ref(gossip_address_map)).get();
```

Comment:
> I know you're just copying code we already had in Enterprise, but this patch made me feel a bit uneasy. First, it seems you removed the "$user" "tenant", which I have vague memory we recently had problems with, and I hope it doesn't break with this patch.

It was the "$maintenance" tenant that we recently had problems with. As for the "$user" tenant, we do have code to handle it, see `messaging_service::scheduling_group_for_isolation_cookie`.

Reply:
> Second, before this series service levels in the master branch were some low-impact feature through which you can ask for different timeouts or something. After this patch, the service levels mean a very different thing - having multiple service levels now adds different scheduling groups, adds more connections, etc. I hope we will document the new implications of having service levels.

@tzach FYI (pinging just in case, but the cover letter should make it rather clear)

Just in case, I'll also leave a link (https://github.com/scylladb/scylla-enterprise/pull/4759#issuecomment-2405531598) to a summary of a discussion that we had some time ago with @avikivity; there was an idea to repurpose NULL shares to make service levels with NULL shares behave as in open source, but we decided against that (would be a breaking change for enterprise users, SLs are not considered popular enough in OSS to justify the work).